app/services/notification_service.rb

```diff
@@ -24,8 +25,12 @@ def set_id
   def notify_users
     player_ids = matching_users.map(&:player_id)
 
-    uri = URI.parse(ENV['ONE_SIGNAL_NOTIFY_URL'])
-    params = target_match_request_params(player_ids)
+    uri = URI.parse(ENV['ONE_SIGNAL_NOTIFY_URL'].to_s)
```

Comment:
Is the `to_s` needed? I think Figaro works only with strings (I'm not 100% sure, take a look into this).
Also I recommend you to take a look into this related to ENVs:
https://stackoverflow.com/questions/47985660/rails-difference-between-env-fetch-and-env
Sometime you want your code to work without the environment variables defined in the `application.yml` and that's fine, but I think this is not the case and helps you not to forget setting the variables also in Heroku.

Reply:
I did not understand the last part @leandrodzp 
Don't I want the URL stored on my env variables?